scripts/build.actions.js

```diff
@@ -33,50 +33,67 @@ class BuildActions extends BaseScript {
    * @memberof DeployActions
    */
   async run (args = [], buildConfig = {}) {
-    if (!this.config.app.hasBackend) throw new Error('cannot build actions, app has no backend')
+    if (!this.config.app.hasBackend) {
+      throw new Error('cannot build actions, app has no backend')
+    }
     const taskName = 'Build actions'
     this.emit('start', taskName)
 
+    // this isn't 100% necessary, everything seems fine without it
     fs.emptyDirSync(this.config.actions.dist)
 
     const build = async (name, action) => {
       const actionPath = this._absApp(action.function)
       const outPath = path.join(this.config.actions.dist, `${name}.zip`)
+      const tempBuildDir = path.join(path.dirname(outPath), `${name}-temp`) // build all to tempDir first
       const actionFileStats = fs.lstatSync(actionPath)
 
+      // make sure temp/ exists
+      fs.ensureDirSync(tempBuildDir)
+
       if (!actionFileStats.isDirectory() && !actionFileStats.isFile()) {
         throw new Error(`${action.function} is not a valid file or directory`)
       }
 
+      // Process include(d) files
+      const includeFiles = await utils.getIncludesForAction(action)
+      includeFiles.forEach(incFile => {
+        const dest = path.join(tempBuildDir, incFile.dest)
+        fs.ensureDirSync(dest)
+        // dest is expected to be a dir ...
+        incFile.sources.forEach(file => {
+          fs.copyFileSync(file, path.join(dest, path.parse(file).base))
+        })
+      })
+
       if (actionFileStats.isDirectory()) {
         // make sure package.json exists OR index.js
         const packageJsonPath = path.join(actionPath, 'package.json')
         if (!fs.existsSync(packageJsonPath)) {
           if (!fs.existsSync(path.join(actionPath, 'index.js'))) {
             throw new Error(`missing required ${this._relApp(packageJsonPath)} or index.js for folder actions`)
           }
-          aioLogger.debug('found an index.js, allowing zip')
+          aioLogger.debug('action directory has an index.js, allowing zip')
         } else {
           // make sure package.json exposes main or there is an index.js
           const expectedActionName = utils.getActionEntryFile(packageJsonPath)
           if (!fs.existsSync(path.join(actionPath, expectedActionName))) {
             throw new Error(`the directory ${action.function} must contain either a package.json with a 'main' flag or an index.js file at its root`)
           }
           // install dependencies
-          await utils.installDeps(actionPath)
+          // await utils.installDeps(actionPath) // TODO: this should be flaggable/avoidable
```

Comment:
should this be in here or just deleted?

Reply:
This is an open question, do we want to run npm i every time we encounter a package.json? 
A typical `npm i` takes 5 seconds even if nothing has changed, and everything is already installed.